tools/rvn/CommandLineApp.cs

```diff
@@ -52,6 +63,209 @@ public static int Run(string[] args)
+            }
+
+            if (File.Exists(parameters.SetupJsonPath) == false)
+            {
+                throw new InvalidOperationException($"-s|--setup-json-path path:{parameters.SetupJsonPath} not found");
+            }
+
+            using (StreamReader file = File.OpenText(parameters.SetupJsonPath))
+            {
+                JsonSerializer serializer = new();
+                setupInfo = (SetupInfo)serializer.Deserialize(file, typeof(SetupInfo));
+            }
+        }
+
+        private static void ValidateSetupInfo(SetupInfo setupInfo, CreateSetupPackageParameters parameters)
+        {
+            if (setupInfo.License == null)
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.License)} must be set");
+            }
+
+            if (setupInfo.License.Keys is null || setupInfo.License.Keys.Any() == false)
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.License.Keys)} must be set");
+            }
+
+            if (string.IsNullOrEmpty(setupInfo.License.Id.ToString()))
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.License.Id)} must be set");
+            }
+
+            if (string.IsNullOrEmpty(setupInfo.License.Name))
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.License.Name)} must be set");
+            }
+
+            if (string.IsNullOrEmpty(setupInfo.Email))
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.Email)} must be set");
+            }
+
+            if (string.IsNullOrEmpty(setupInfo.Domain))
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.Domain)} must be set");
+            }
+
+            if (string.IsNullOrEmpty(setupInfo.RootDomain))
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.RootDomain)} must be set");
+            }
+
+            if (setupInfo.NodeSetupInfos is null || setupInfo.NodeSetupInfos.Any() == false)
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.NodeSetupInfos)} must be set");
+            }
+
+            foreach (var tag in setupInfo.NodeSetupInfos.Keys.Where(tag => IsValidNodeTag(tag) == false))
+            {
+                throw new InvalidOperationException($"{tag} - node tag must contain only capital letters. Maximum length should be up to 4 characters");
+            }
+
+            foreach (var nodeInfoNode in setupInfo.NodeSetupInfos.Values)
+            {
+                if (nodeInfoNode?.Addresses is null)
+                {
+                    throw new InvalidOperationException($"Addresses must be set inside {nameof(setupInfo.NodeSetupInfos)}");
+                }
+
+                if (nodeInfoNode.Port == 0)
+                {
+                    nodeInfoNode.Port = Raven.Client.Constants.Network.DefaultSecuredRavenDbHttpPort;
```

Comment:
Since this method has side-effects like this we could rename it to "ValidateSetupInfoAndSetDefaultSetupParametersIfNeeded"?

Reply:
done